File: LAB3/src/Domain/sarcina.java
package Domain;

public class sarcina extends UniqueById<Integer> {

	private String descriere;
	
	public sarcina(){
		this(-1, "");
	}
	
	public sarcina (Integer id , String desc){
		this._id=id;
		this.descriere=desc;
	}
	
	public Integer get_id(){
		return this._id;
	}
	
	public void  set_id(Integer a ){
		this._id=a;
	}
	
	public String get_descriere (){
		return this.descriere;
	}
	
	public void set_descriere(String desc){
		this.descriere=desc;
	}
	
	public String toString(){
		return _id +" "+ this.descriere;
	}
}

File: lab1cSharp/Domain/sarcina.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab1cSharp.Domain
{
    public class sarcina:hasId<int>
    {
        
        private string _descriere;
        public string  descriere { get { return _descriere; } set { _descriere = value; } }
        public sarcina()
        {
            _id = -1;
            _descriere = "";
        }

        public sarcina(int id, string desc):base(id)
        {
            _descriere = desc;          
        }

        public override string ToString()
        {
            return base.ToString() + " " + _descriere.PadRight(15);
        }

        public override bool Equals(object obj)
        {
            if (obj == null)
                return false;
            if (!(obj is sarcina))
                return false;
            sarcina aux = (sarcina)obj;
            return _id == aux.id && _descriere ==aux.descriere;
        }

        public string export(string separator)
        {
            return id.ToString() + separator +descriere+separator;
        }

    }

}
